unleash-yggdrasil/src/lib.rs

```diff
@@ -345,25 +345,35 @@ impl EngineState {
         })
     }
 
-    fn enabled(&self, toggle: &CompiledToggle, context: &Context) -> bool {
-        let enriched_context = EnrichedContext::from(context.clone(), toggle.name.clone());
+    fn enabled(
+        &self,
+        toggle: &CompiledToggle,
+        context: &Context,
+        external_values: Option<HashMap<String, bool>>,
+    ) -> bool {
+        let enriched_context =
+            EnrichedContext::from(context.clone(), toggle.name.clone(), external_values);
         toggle.enabled
             && self.is_parent_dependency_satisfied(toggle, context)
             && (toggle.compiled_strategy)(&enriched_context)
     }
 
-    pub fn resolve_all(&self, context: &Context) -> Option<HashMap<String, ResolvedToggle>> {
+    pub fn resolve_all(
+        &self,
+        context: &Context,
+        external_values: Option<HashMap<String, bool>>,
+    ) -> Option<HashMap<String, ResolvedToggle>> {
         self.compiled_state.as_ref().map(|state| {
             state
                 .iter()
                 .map(|(name, toggle)| {
-                    let enabled = self.enabled(toggle, context);
+                    let enabled = self.enabled(toggle, context, external_values.clone());
```

Comment:
Should we borrow instead of cloning? I'm thinking this should probably be immutable at this point

Reply:
Borrow vs owned is more about memory management than mutability but this is good feedback, this doesn't need to be a clone at this point. I've made this propagate via a borrow all the way down to the where the enriched context is build, which does require an owned type